doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+* We'll only support one command per commandline. This will be one of the
+  following list:
+  - `help`
+  - `version`
+  - `new-tab`
+  - `open-settings`
+* We'll need to make sure that we process commandlines with escaped semicolons
+  in them the same as we will in 2.0. Users will need to escape `;` as `\;` in
+  1.0, even if we don't support multiple commands in 1.0.
+* If users don't provide a command, we'll assume the command was `new-tab`. This will be in-line with
+
+
+#### Sample 1.0 Commandlines
+
+```sh
+# display the help message
+wt help
+wt --help
+wt -h
+wt -?
+
+# Display version info for the Windows Terminal
+wt version
+wt --version
+wt -v
+
+# Runs the user's default profile in a new tab, running cmd.exe
+wt cmd.exe
+
+# Runs the user's "Windows Powershell" profile in a new tab
+wt new-tab --profile "Windows Powershell"
+wt --profile "Windows Powershell"
+
+# run "my-commandline.exe with some args" in a new tab
+wt new-tab my-commandline.exe with some args
+wt my-commandline.exe with some args
+
+# run "my-commandline.exe with some args and a ; literal semicolon" in a new
+#  tab, and in another tab, run "another.exe running in a second tab"
+wt my-commandline.exe with some args and a \; literal semicolon
+
+# Start the default profile in directory "c:/Users/Foo/dev/MyProject" (user story 8)
+wt new-tab --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -d "c:/Users/Foo/dev/MyProject"
+
+# Runs the user's "Windows Powershell" profile in a new tab in directory
+#  "c:/Users/Foo/dev/MyProject" (user story 2, 8)
+wt new-tab --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -p "Windows Powershell" -d "c:/Users/Foo/dev/MyProject"
+
+```
+
+## Implementation Details
+
+TODO: I'm leaving this empty for the time being. I want to get some feedback on
+the proposal in this state, before starting work on implementation details. This
+style of arguments might be controversial, so I want to make sure we settle on a
+syntax before I get too far into the details of parsing and passing these args
+around.
+
+## Capabilities
+
+### Accessibility
+
+As a commandline feature, the accessibility of this feature will largely be tied
+to the ability of the commandline environment to expose accessibility
+notifications. Both `conhost.exe` and the Windows Terminal already support
+basica accessibility patterns, so users using this feature from either of those
+terminals will be reliant upon their accessibility implementations.
+
+### Security
+
+As we'll be parsing user input, that's always subject to worries about buffer
+length, input values, etc. Fortunately, most of this should be handled for us by
+the operating system, and passed to us as a commandline via `winMain` and
+`CommandLineToArgvW`. We should still take extra care in parsing these args.

```

Comment:
We should probably consider using arg-parsing NuGet that is licensing-compatible with us. And we should fuzz it with our fuzzer if it's not already fuzzed as a security consideration.

Reply:
Yea the "how" protion I was kinda putting off till I had more consensus on the syntax. If we come to enough of a consensus on design, I'll do the research to find the appropriate parser.

Reply:
(this has been added now)